### DISCUSSION TITLE
Update 2022-08-24

### DISCUSSION THREAD

**Discussion by kaveh808** [kaveh808/kons-9#35] [Announcements]
We now have an updated main branch which runs on MacOS and Linux.

Feel free to use this version for your testing and development. I will be making updates and getting more of the demos to work over the next few days.

**Comment by JMC-design** [comment]
Any reason there isn't a simple (gl:viewport 0 0 w h)  before the draw-scene call in the resize callback?
Seems to work fine with it. https://youtube.com/clip/UgkxkLzVHNSg6QVuML0nYs3CVTueCpujFOvT
This on linux/sbcl.

**Comment by kaveh808** [comment]
The current code works fine on MacOS. I couldn't make out your video.

If I add the call to gl:viewport, the viewport dimensions are half of the window's. Maybe MacOS counting 2 pixels as 1 in high-res displays?

**Comment by JMC-design** [comment]
ah, the video just shows it resizing.  Without the call no resizing happens at all on my end.
I wonder if that's a linux thing, or something to do with my window-manager.

**Comment by JMC-design** [comment]
I checked on ubuntu's compositing window manager and it doesn't resize there either.
Could just stick in a +linux in there, though now I'm curious what happens in Windows.
Let me reboot and try in windows.

**Reply by JMC-design** [reply]
I don't understand the stupidity of Windows. Doesn't microsoft own github?  Same as yesterday I can't download or see the recent updates on windows. 
Seems cl-glfw3 has no installation instructions for windows and might not work here, no idea where it expects to find the lib.
I'll do a glop thing tomorrow and copy it to windows and see if same happens.

**Reply by mikelevins** [reply]
Generally the easiest way to deal with shared libraries on Windows is to put them in the same directory as the executable that loads them. In the case of this project that should be the directory that contains the sbcl executable.

**Reply by JMC-design** [reply]
and i think this is why Shinmera wrote Deploy.

**Comment by kaveh808** [comment]
All demos now work.

**Reply by mikelevins** [reply]
Nice!

**Reply by kaveh808** [reply]
Fixed a missing function bug.
